sources/constraint-definition.asciidoc

```diff
@@ -455,7 +460,18 @@ if [classname]`@FrenchZipCode` is defined as
 include::{spec-examples-source-dir}org/beanvalidation/specexamples/constraintdefinition/constraintcomposition/overriding/interpreted/FrenchZipCode.java[tags=include]
 ----
 
-[tck-testable]#If a constraint is used more than once as a composing constraint, the multi value constraints model as described in <<constraintsdefinitionimplementation-multipleconstraints>> is used.# [tck-testable]#To select a specific composing constraint, [methodname]`OverridesAttribute.constraintIndex` is used. It represents the constraint index in the [methodname]`value` array.# [tck-testable]#If `index` is undefined, the single constraint declaration is targeted.#
+[tck-testable]#If a constraint is used more than once as a composing constraint, the multi value constraints model as described in <<constraintsdefinitionimplementation-multipleconstraints>> is used.#
+
+To select a specific composing constraint, [methodname]`OverridesAttribute.constraintIndex` is used.
+[tck-testable]#If the composing constraints are directly given on the composing constraint, `constraintIndex` refers to the left-to-right order of the constraints of this type in which they are given on the composing constraint.#
+[tck-testable]#If the composing constraints are specified using their corresponding `List` annotation, `constraintIndex` refers to the index within the `value` array.#
+[tck-testable]
+--
+A composing constraint must not be given directly on the composing constraint and using the corresponding `List` annotation at the same time.
```

Comment:
Actually it is allowed to have one instance of the repeatable annotation type and one instance of the container annotation. See the second example in the section you linked:

> This rule is designed to allow the following code:
>
> @Foo(1) @FooContainer({@Foo(2)})
> class A {}

But we can determine that as `AnnotatedElement#getDeclaredAnnotations()` returns the _directly present_ annotations, e.g. `@Pattern` and `@Pattern.List`. In this case an exception would have to be thrown.